src/Application.tsx

```diff
@@ -1,35 +1,59 @@
-import React, { Suspense } from "react";
+import React, { Suspense, useEffect } from "react";
 import {
   Route,
-  Switch,
   RouteComponentProps,
+  Switch,
+  withRouter,
 } from "react-router-dom";
-import routes from "src/common/routes";
 
-const Application: React.FunctionComponent<{}> = (props) => {
+import { guestRoutes, userRoutes } from "src/common/routes";
+import { routes } from "./common/constants/pageRoutes";
+import { useAuth } from "./context/GlobalContext";
+
+const Application: React.FC<
+  RouteComponentProps<{}, {}, { from: { pathname: string } }>
+> = ({ history }) => {
+  const { state } = useAuth();
+
+  useEffect(() => {
```

Comment:
[Optional]
I think this can be pulled out into a reusable hook which can be consumed via each route.
That way a route can decide if it wan't to redirect the user to the auth page.

Reply:
have made the hook but not using it at the component level because we already have a guest and authenticated routes separately using this hook at each route level would just increase checks which can be avoided if we only use it once at the time of application load.